src/main/java/com/webdev/cosmo/cosmobackend/service/user/UserService.java

```diff
@@ -0,0 +1,51 @@
+package com.webdev.cosmo.cosmobackend.service.user;
+
+import com.webdev.cosmo.cosmobackend.service.api.User;
+import com.webdev.cosmo.cosmobackend.service.common.mapper.UserMapper;
+import com.webdev.cosmo.cosmobackend.service.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.openapitools.model.UserModel;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class UserService {
+
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    public UserModel save(User user) {
+        User savedUser = userRepository.save(user);
+
+        log.info("Saved User: " + savedUser);
+
+        return userMapper.mapToModel(savedUser);
+    }
+
+    public UserModel findByEmail(String email) {
+        User user = userRepository.findByEmail(email);
```

Comment:
When Optional will be introduced this will change. Please see Optional.ofNullable(sth).map(sth).orElseThrow()